src/Modules/OrchardCore.Commerce/Endpoints/Extentions/HttpContextExtensions.cs

```diff
@@ -0,0 +1,21 @@
+using Microsoft.AspNetCore.Http;
+using OrchardCore.Environment.Shell.Scope;
+
+namespace OrchardCore.Modules;
+
+public static class HttpContextExtensions
+{
+    /// <summary>
+    /// Makes <see cref="HttpContext.RequestServices"/> aware of the current <see cref="ShellScope"/>.
+    /// </summary>
+    public static HttpContext UseShellScopeServices(this HttpContext httpContext)
+    {
+        httpContext.RequestServices = new ShellScopeServices(httpContext.RequestServices);
+        return httpContext;
+    }
```

Comment:
Please delete this, since it's not used and it's already available [in the current OC release](https://github.com/OrchardCMS/OrchardCore/blob/v1.8.0/src/OrchardCore/OrchardCore.Abstractions/Modules/Extensions/HttpContextExtensions.cs) anyway.